xsct.c

```diff
@@ -0,0 +1,234 @@
+ *
+ * Compile the code using the following command:
+ * cc -std=c99 -O2 -I /usr/X11R6/include sct.c -o xsct -L /usr/X11R6/lib -lX11 -lXrandr
+ *
+ * Original code published by Ted Unangst:
+ * http://www.tedunangst.com/flak/post/sct-set-color-temperature
+ *
+ * Modified by Fabian Foerg in order to:
+ * - compile on Ubuntu 14.04
+ * - iterate over all screens of the default display and change the color
+ *   temperature
+ * - fix memleaks
+ * - clean up code
+ * - return EXIT_SUCCESS
+ *
+ * Public domain, do as you wish.
+ *
+ * gcc -Wall -Wextra -Werror -pedantic -std=c99 -O2 -I /usr/X11R6/include sct.c -o xsct -L /usr/X11R6/lib -lX11 -lXrandr -lm -s
+ *
+ */
+
+#include <X11/Xatom.h>
+#include <X11/Xlib.h>
+#include <X11/Xproto.h>
+#include <X11/extensions/Xrandr.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+static void usage(char * pname)
+{
+    printf("Xsct (1.5)\n"
+           "Usage: %s [options] [temperature]\n"
+           "\tIf the argument is 0, xsct resets the display to the default temperature (6500K)\n"
+           "\tIf no arguments are passed, xsct estimates the current display temperature\n"
+           "Options:\n"
+           "\t-v, --verbose \t xsct will display debugging information\n"
+           "\t-h, --help \t xsct will display this usage information\n", pname);
+}
+
+#define TEMPERATURE_NORM    6500
+#define TEMPERATURE_ZERO    700
+#define GAMMA_MULT          65535.0
+// Approximation of the `redshift` table from
+// https://github.com/jonls/redshift/blob/04760afe31bff5b26cf18fe51606e7bdeac15504/src/colorramp.c#L30-L273
+// without limits:
+// GAMMA = K0 + K1 * ln(T - T0)
+// Red range (T0 = TEMPERATURE_ZERO)
+// Green color
+#define GAMMA_K0GR          -1.47751309139817
+#define GAMMA_K1GR          0.28590164772055
+// Blue color
+#define GAMMA_K0BR          -4.38321650114872
+#define GAMMA_K1BR          0.6212158769447
+// Blue range  (T0 = TEMPERATURE_NORM - TEMPERATURE_ZERO)
+// Red color
+#define GAMMA_K0RB          1.75390204039018
+#define GAMMA_K1RB          -0.1150805671482
+// Green color
+#define GAMMA_K0GB          1.49221604915144
+#define GAMMA_K1GB          -0.07513509588921
+
+static double DoubleTrim(double x, double a, double b)
+{
+    double buff[3] = {a, x, b};
+    return buff[ (int)(x > a) + (int)(x > b) ];
+}
+
+static int get_sct_for_screen(Display *dpy, int screen, int fdebug)
+{
+    Window root = RootWindow(dpy, screen);
+    XRRScreenResources *res = XRRGetScreenResourcesCurrent(dpy, root);
+
+    int temp = 0, n, c;
+    double t = 0.0;
+    double gammar = 0.0, gammag = 0.0, gammab = 0.0, gammam = 1.0, gammad = 0.0;
+
+    n = res->ncrtc;
+    for (c = 0; c < n; c++)
+    {
+        RRCrtc crtcxid;
+        int size;
+        XRRCrtcGamma *crtc_gamma;
+        crtcxid = res->crtcs[c];
+        crtc_gamma = XRRGetCrtcGamma(dpy, crtcxid);
+        size = crtc_gamma->size;
+        gammar += (crtc_gamma->red[size - 1]);
```

Comment:
```suggestion
        gammar += crtc_gamma->red[size - 1];
```
same for the two lines below

Reply:
Resolve https://github.com/faf0/sct/pull/10/commits/8774dceb71da59fd952f6209d3a0d63496776fd3